packages/x-priority-queue/src/x-priority-queue.ts

```diff
@@ -0,0 +1,241 @@
+import { AnyFunction } from '@empathyco/x-utils';
+
+/**
+ * An XPriorityQueueNode object is a representation of a structure containing a parametrized key, a
+ * priority number and metadata record. By default, the key is a string.
+ *
+ * @public
+ */
+export class XPriorityQueueNode<Key = string> {
+  /**
+   * The key to store the element in the queue.
+   *
+   * @public
+   */
+  public readonly key: Key;
+  /**
+   * The number used to sort the elements in the queue.
+   *
+   * @public
+   */
+  public readonly priority: number;
+  /**
+   * The extra data to store in the queue associated with a key and priority pair. Optionally, a
+   * `boolean` `replaceable` key can be used in order to make the node replaceable in the queue.
+   * Being replaceable means that if a new element is pushed into the queue with the same key, the
+   * existing one will be removed.
+   *
+   * @public
+   */
+  public readonly metadata: Record<string, unknown>;
```

Comment:
What would you think of defining a type for the `metadata`? Just to add the restriction that could have a `replaceable` property (optionally) and it would be a `boolean`, so we prevent that property to be passed with an unexpected value.